Route refresh: added `incidents`

RingerJK opened the pull request:
~depends on https://github.com/mapbox/mapbox-java/pull/1421~

require NN API: update incidents (⚠️ landed in 99.0.0)

<!--
---------- CHECKLIST ----------
1. Add related labels (`bug`, `feature`, `new API(s)`, `SEMVER-MAJOR`, `needs-backporting`, etc.).
1. Adda a changelog entry under `Unreleased` tag or a `skip changelog` label if not applicable.
1. Update progress status on the project board.
1. Request a review from the team, if not a draft.
1. Add targeted milestone, when applicable.
1. Create ticket tracking addition of public documentation pages entry, when applicable.
-->

VysotskiVadim commented:
Please invalidate incidents in the same way as annotations if https://github.com/mapbox/mapbox-navigation-android/pull/5767 lands first.

RingerJK commented on build.gradle:
```diff
@@ -58,7 +58,7 @@ allprojects {
     // if you need to use snapshots while development, just set `addSnapshotsRepo` to true manually
     def addSnapshotsRepo = project.hasProperty('ALLOW_SNAPSHOT_REPOSITORY') ? project.property('ALLOW_SNAPSHOT_REPOSITORY') : (System.getenv("ALLOW_SNAPSHOT_REPOSITORY")?.toBoolean() ?: false)
-    if (addSnapshotsRepo) {
+    if (true) {
```
uses to allow snapshots only, will be reverted when the next mapbox-java will be released

Zayankovsky replied:
v6.5.0-beta.5 should be available already.

VysotskiVadim commented on libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/NavigationRouteExTest.kt:
```diff
@@ -0,0 +1,200 @@
+class NavigationRouteExTest {
+
+    @Test
+    fun `extension NavigationRoute refreshRoute`() {
```
No actions required, just a feedback from me. I find this "parametrised" tests hard to read. I spent 20 minutes and didn't really get all the scenarios. Usually if you write a test per scenario, tests are getting smaller and easier to understand

RingerJK replied:
thanks for the feedback, parametrized tests are helpful for different cases in the same scenario (like here), not setting up an environment for each test. I'd add more docs for current tests, to make it readable

LukasPaczos reviewed (commented):
@RingerJK we also need to extract the road objects from `RouteInterface#routeInfo#alerts` after calling `Navigator#refreshRoute` and deliver these refreshed alerts in the `RoadObjectsOnRouteObserver`.

LukasPaczos commented on CHANGELOG.md:
```diff
@@ -8,6 +8,7 @@ Mapbox welcomes participation and contributions from everyone.
 - Added `TollCollection#name` field which contains a name of the toll collection point, when available. [#5784](https://github.com/mapbox/mapbox-navigation-android/pull/5784)
 - Added `ReplayRouteOptions#frequency` to allow adjusting the simulated location frequency. [#5724](https://github.com/mapbox/mapbox-navigation-android/pull/5724)
 - Added `MapboxReplayer#eventRealtimeOffset` to allow adjusting simulated locations timestamps for more accurate replays. [#5748](https://github.com/mapbox/mapbox-navigation-android/pull/5748)
+- Added update of `incidents` for route refresh. For now `incidents` are refreshed along a route. [#5749](https://github.com/mapbox/mapbox-navigation-android/pull/5749)
```
Please also add an instrumentation test for this in `com.mapbox.navigation.instrumentation_tests.core.RouteRefreshTest`.

RingerJK replied:
We usually start changelog lines from Added/Updated/Removed. I'd change to align 
```suggestion
- Updated feature `route refresh`: now also supports refreshing `incidents` along the route found in `RouteLeg#incidents` and `RoadObjectsOnRouteObserver`. [#5749](https://github.com/mapbox/mapbox-navigation-android/pull/5749)
```

RingerJK commented:
- added refresh route tests to `com.mapbox.navigation.instrumentation_tests.core.RouteRefreshTest` (in the same test added check `RoadObjectsOnRouteObserver` data);
- removed refresh route tests from `com.mapbox.navigation.navigator.internal.NavigatorTest` as it duplicates with the above point;
- added description to parametrized test wrappers of `com.mapbox.navigation.base.internal.route.NavigationRouteExTest`;
- fixed changelog.
